src/EventListener/ReadListener.php

```diff
@@ -90,6 +95,10 @@ public function onKernelRequest(GetResponseEvent $event)
         try {
             $identifiers = $this->extractIdentifiers($request->attributes->all(), $attributes);
 
+            if (null !== $this->dispatcher) {
+                $this->dispatcher->dispatch(PreReadEvent::NAME, new PreReadEvent($data));
```

Comment:
It's really weird to have data in Pre Read. I don't think it makes sense...

Reply:
Perhaps what we could do is to keep a reference to the event object, then check if the data is already set on the event. If it is set, we should not override it.